### ISSUE TITLE
curator.utils.rollable_alias falsely reports aliases with multiple indices as rollable

### ISSUE THREAD

**Issue by Battleroid** [elastic/curator#1276]
## To submit a bug or report an issue
If you have an alias that points to multiple indices the helper function `rollable_alias` falsely reports the alias as rollable.

### Expected Behavior
Should not report it as rollable.

### Actual Behavior
It reports it as rollable.

## Context (Environment)
We use the `rollable_alias` func to determine which aliases are rollable to automate rollover requests for our clusters.

**Comment by untergeek** [comment]
So what you're saying is that [this code block](https://github.com/elastic/curator/blob/master/curator/utils.py#L96-L98) isn't working as expected?

**Comment by Battleroid** [comment]
I might have spoke too soon, it works as expected, but it gets weird when we have two aliases that point to the same index, each being the same index. For example:

```
a                  i
mysql-slow-session mysql-slow-session-2018.09.18-000003
request-id         mysql-slow-session-2018.09.18-000003
```

This is due to the fact that the template for mysql-slow-session maps to the request-id alias on creation.

**Comment by untergeek** [comment]
Sorry for the long delay in response. It's been a _very_ busy last quarter for me. I'm redoubling my efforts on Curator to catch up on my issues. 

## TL;DR

Don't put the rollover alias name in the index template.

## Longer, full explanation with examples

Your error yields a nugget, I just had to scroll to the right a bunch to see it:

```
found duplicated alias [[request-id]] in index template [mysql-slow-session]
```

What this means is that you are assigning `request-id` in the index template. This is not required with the Rollover API, and is the cause of the error message.

I replicated this behavior by putting this in a template called `test`:

```
PUT _template/test
{
    "order": 0,
    "index_patterns": [
      "test-*"
    ],
    "settings": {
      "index": {
        "number_of_shards": "1",
        "number_of_replicas": "0"
      }
    },
    "mappings": {},
    "aliases": {
        "foobar": {},
        "testalias": {}
    }
}
```

Then creating a new index that _should_ be rollable, like `index-000001`, which would be assigned these two aliases:

```
$ curl localhost:9200/_cat/aliases
foobar    test-000001 - - -
testalias test-000001 - - -
```

And when you try to call a rollover and the new index has the rollover alias already defined in the template (as above)?

```
POST testalias/_rollover
{
    "conditions": {
        "max_age": "1s"
    }
}
```
Gives this output:
```
{
    "error": {
        "root_cause": [
            {
                "type": "illegal_argument_exception",
                "reason": "Rollover alias [testalias] can point to multiple indices, found duplicated alias [[testalias, foobar]] in index template [test]"
            }
        ],
        "type": "illegal_argument_exception",
        "reason": "Rollover alias [testalias] can point to multiple indices, found duplicated alias [[testalias, foobar]] in index template [test]"
    },
    "status": 400
}
```
So, the fix is to remove the rollover alias from the template, so it looks like this at the end:
```
    "aliases": {
        "foobar": {}
    }
```
I reapply the template with this change, _et voila,_
```
{
    "acknowledged": true,
    "shards_acknowledged": true,
    "old_index": "test-000001",
    "new_index": "test-000002",
    "rolled_over": true,
    "dry_run": false,
    "conditions": {
        "[max_age: 1s]": true
    }
}
```

And here are my aliases:

```
$ curl localhost:9200/_cat/aliases
foobar    test-000001 - - -
foobar    test-000002 - - -
testalias test-000002 - - -
```
Note that `foobar` (my analog to your `mysql-slow-session` alias) still points to both indices, while `testalias` (our rollover alias name), points only to the rollover index.  If I run it again...

```
$ curl localhost:9200/_cat/aliases
foobar    test-000001 - - -
foobar    test-000003 - - -
testalias test-000003 - - -
foobar    test-000002 - - -
```

To recap, the solution is to remove the rollover alias name from the template. If you had to do this over again, you would add the template first (with only the carry-over alias) and manually add the rollover alias at index creation time like:
```
PUT index-000001
{
  "aliases": {
    "testalias": {}
  }
}
```
